### PR TITLE
Advanced Register Challenge

### PR DESCRIPTION
**Pull request by Light-it-luc**
## Description ⭐

Modified the codebase to use React Hook Form and Zod for validating the register form. The `useFormInput` hook was deleted, as well as the validator functions in `constants/inputValidations`. The validator logis still lives in `constants/inputValidations` but now a zod schema is defined (with the added bonus that all validator logic is in the same place, while before the `confirmPassword` validation logic was in the register form).

## Emoji Guide

**For reviewers: Emojis can be added to comments to call out blocking versus non-blocking feedback.**

E.g., Praise, minor suggestions, or clarifying questions that don’t block merging the PR.

> 🟢 Nice refactor!
> 🟡 Why was the default value removed?

E.g., Blocking feedback must be addressed before merging.

> 🔴 This change will break something important

|              |                |                                     |
| ------------ | -------------- | ----------------------------------- |
| Blocking     | 🔴 ❌ 🚨       | RED                                 |
| Non-blocking | 🟡 💡 🤔 💭    | Yellow, thinking, etc               |
| Praise       | 🟢 💚 😍 👍 🙌 | Green, hearts, positive emojis, etc |

## Links

- [Git Flow](https://lightit.slite.com/app/docs/SC8usN2Ju)
- [Handbook of good practices for reviewers in Code Reviews](https://lightit.slite.com/app/docs/ddNGohWthVB3fO)

### CODE REVIEW

File: 3-Register/src/constants/inputValidations.ts
Comment (sgoycoechea-lightit) [inline 3-Register/src/constants/inputValidations.ts:1]: We should rename this file and move it somewhere else.

Maybe we could name it `registerSchema.ts` and move it to a folder `RegisterForm` inside `components`, next to `RegisterForm.tsx`.

File: 3-Register/src/constants/inputValidations.ts [hunk 1]
```diff
@@ -1,65 +1,37 @@
...
-    error: "Password must have a special character.",
-  },
-];
+import { z } from "zod";
+
+const availableDomains = ["@gmail.com", "@hotmail.com", "@lightit.io"];
+const emailDomainIsValid = (email: string) =>
+  availableDomains.some((domain) => email.endsWith(domain));
+
+export const RegisterSchema = z
```
Comment (sgoycoechea-lightit) [inline 3-Register/src/constants/inputValidations.ts]: Capitalization is only for components and types, this is a variable.

```suggestion
export const registerSchema = z
```

File: 3-Register/src/components/RegisterForm.tsx [hunk 2]
```diff
@@ -9,95 +8,59 @@ import {
...
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<RegisterFormType>({ resolver: zodResolver(RegisterSchema) });
 
   return (
-    <form action="" method="POST" className="flex flex-col gap-4">
+    <form
+      action=""
+      method="POST"
```
Comment (sgoycoechea-lightit) [inline 3-Register/src/components/RegisterForm.tsx]: I don't think this is needed.

```suggestion
```

File: 3-Register/src/constants/inputValidations.ts
Comment (Light-it-luc) [reply 3-Register/src/constants/inputValidations.ts:1]: Ok good. Should I move the `RegisterForm` component to that folder as well?

Comment (sgoycoechea-lightit) [reply 3-Register/src/constants/inputValidations.ts:1]: Yeah, you can rename it to `index.tsx` or keep it with the name

**Review by sgoycoechea-lightit — APPROVED** [review]

**Pull request merged**